### ISSUE TITLE
Meteor: 3.0.0-rc1 Package: allow-deny -> client side collection insert/update/remove not working as expected

### ISSUE THREAD

**Issue by acemtp** [meteor/meteor#13143]
When executing the following line on the client with meteor 3, it should work since minimongo doesn't require async function calls.

`Meteor.users.update(Meteor.userId(), { $set: { 'profile.ok': 1 }})`

It generates this error on the server:

```
 Exception while invoking method '/users/update' Error: findOne +  is not available on the server. Please use findOneAsync() instead.
     at Object.ret.<computed> (packages/mongo/remote_collection_driver.js:52:15)
     at Object.<anonymous> (packages/matb33:collection-hooks/findone.js:27:22)
     at Object.collection.<computed> [as findOne] (packages/matb33:collection-hooks/collection-hooks.js:116:21)
     at CollectionPrototype._validatedUpdate (packages/allow-deny/allow-deny.js:485:32)
     at MethodInvocation.m.<computed> (packages/allow-deny/allow-deny.js:193:46)
```

The reason is that in allow-deny.js of the AllowDeny package, the old method _validatedUpdate() is called but contains a findOne that is not Async.

Current Fix:

On our server code, we added this lines to override the deprecated function

```
if (Meteor.isServer) {
  Mongo.Collection.prototype._validatedInsert = Mongo.Collection.prototype._validatedInsertAsync;
  Mongo.Collection.prototype._validatedUpdate = Mongo.Collection.prototype._validatedUpdateAsync;
  Mongo.Collection.prototype._validatedRemove = Mongo.Collection.prototype._validatedRemoveAsync;
}
```

Solution 1:

Update the code of _validatedInsert _validatedUpdate _validatedRemove in allow-deny.js to handle async methods

Solution 2 (better):

remove _validatedInsert _validatedUpdate _validatedRemove function and always use _validatedInsertAsync _validatedUpdateAsync _validatedRemoveAsync in both case update and updateAsync.

**Comment by nachocodoner** [comment]
Thank you for the report.

It's surprising we have not run into this issue in our automated test coverage, but we'll make sure to add it and include the fix in our nexts RCs.

**Comment by denihs** [comment]
Yes, it is strange... Anyways, let's fix it!

**Comment by nachocodoner** [comment]
FYI, this issue is being addressed [in this PR](https://github.com/meteor/meteor/pull/13168) and is scheduled for release in the next RC.3.

If there are other issues with the allowdeny package, this is the perfect opportunity to address them. Report or refer to any additional issues directly here.

**Comment by acemtp** [comment]
Thanks, we'll try as soon as the rc3 is released!

**Comment by denihs** [comment]
Hey @acemtp we have now released version [`3.0-rc.4`](https://forums.meteor.com/t/meteor-3-0-rc-4-is-out/61742). Can you test this again and make sure it works for you?

**Comment by acemtp** [comment]
Excellent, seems to work fine, you can close the issue.

In the doc below you still have insert/update/remove function in allow deny, but in which cases they are called since now calling collection.insert or collection.insertAsync in client always call the allow.insertAsync function?

https://v3-docs.meteor.com/api/collections.html#Mongo-Collection-allow
